### ISSUE TITLE
getS3Objects with prefix checks the whole object name against the prefix

### ISSUE THREAD

**Issue by kierachell** [adobe/S3Mock#97]
### Test Case:

1. Given an s3Mock with a bucket ful'o'files and no directories

1. The base case with a `ListObjectsV2Request` will list all the files in the bucket.

    ```
     fandom
    country~1
    fanciful~1
    fanciful~2
    fanciful~3
    fanciful~4
    bar~1
    bar~2
    bar~3
    bar~4
    bar~5
    biz
    bonkers
    ```

1. However, if one is to use a `withPrefix` matching a part of a given obejct's name such as `f` or `bar~` or `fanciful~`:

    ```
    ListObjectsV2Request req = new ListObjectsV2Request()
      .withBucketName(bucketName)
      .withMaxKeys(50)
      .withPrefix(prefix);

    ListObjectsV2Result result = s3.listObjectsV2(req);;
    ```

1. You will get an empty list. Whereas the real S3 client will return non-zero amount of objects.

### The problem:

In the [`FileStore`](https://github.com/adobe/S3Mock/blob/master/server/src/main/java/com/adobe/testing/s3mock/domain/FileStore.java) class we have:

`return isEmpty(prefix) || (null != p && p.startsWith(prefix));`

- According to https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#startsWith(java.nio.file.Path):

```
On UNIX for example, the path "foo/bar" starts with "foo" and "foo/bar". It does not start with "f" or "fo".
```

So this is trying to check if the given path `p` matches the whole prefix up to `/` whereas the actual S3 implementation is closer to `p.toString().startsWith(prefix)`.

**Comment by agudian** [comment]
Thanks for the detailed bug report! 

Would you be open to create a PR with a fix, as you already found the right spot?

**Comment by kierachell** [comment]
I will as soon as 1-2 other people confirm that my assumption/understanding on the last line of the issue is correct.

**Comment by agudian** [comment]
@hennejg do you want to add something to this?

**Comment by agudian** [comment]
Fixed through #98 by @kierachell 🥇 

I've triggered a release build... :)
